### PR TITLE
Pino - update pretty formatter option

### PR DESCRIPTION
**Pull request by Paul-Isache**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
-  [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/pinojs/pino/blob/master/docs/API.md#user-content-parameters-1
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: pino/index.d.ts [hunk 1]
```diff
@@ -10,9 +10,18 @@ import stream = require('stream');
 declare function P(optionsOrStream?: P.LoggerOptions | stream.Writable | stream.Duplex | stream.Transform): P.Logger;
 declare function P(options: P.LoggerOptions, stream: stream.Writable | stream.Duplex | stream.Transform): P.Logger;
 
+interface IPinoLog {
```
Comment (DanielRosenwasser) [inline pino/index.d.ts]: Move this into the namespace so it can be referenced.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: pino/index.d.ts [hunk 2]
```diff
@@ -10,9 +10,18 @@ import stream = require('stream');
...
+    level: number
+    time: string
+    msg: string
+    v: number
+}
+
 declare namespace P {
 
-    function pretty(opts?: {timeTransOnly?: boolean, levelFirst?: boolean, formatter?: (str: string) => void}): stream.Transform;
+    function pretty(opts?: {timeTransOnly?: boolean, levelFirst?: boolean, formatter?: (log: IPinoLog) => string}): stream.Transform;
```
Comment (psnider) [inline pino/index.d.ts]: The [documentation](https://github.com/pinojs/pino/blob/master/docs/API.md#pretty) states:
> formatter (function): a custom function to format the line, is passed the JSON object as an argument and should return a string value.

and the proposed change doesn't seem to agree.
I think ```formatter?: (log: {}) => string``` should work.

Comment (Paul-Isache) [inline pino/index.d.ts]: I have created the interface to have a better look on the object that is passed to the function.

File: pino/index.d.ts (hunk 1 above)
Comment (Paul-Isache) [inline pino/index.d.ts]: I moved the interface inside the namespace

**Comment by Paul-Isache** [comment]
I have made the changes. Is it possible to have another look and maybe merge it?

File: pino/index.d.ts (hunk 2 above)
Comment (psnider) [inline pino/index.d.ts]: I think the IPinoLog interface isn't quite correct, and I don't think it is required.
When I run pretty(), and JSON.stringify the passed object, I find:
> time: number

I still prefer the simple typing of **{}** as in my suggestion.

Also keep in mind that there will be many more fields in the object, that is, all of those that are dynamically added from the type to be logged.

@DanielRosenwasser 
There used to be a guideline for writing declarations that advised against typing parts of the interface that aren't documented, but I can no longer find this.
The type of the argument to *formatter* is only documented as an object, or **{}**

**Pull request merged**